README.md
# Mob Boss v1.02 (Release)

Milestone Project #1 for UNLV Software Development Bootcamp.

This is my second game developed for the Milestone Project. After running into a wall shortly before the due date, project #1, Tomb Divers, was shelved for a later date.

## Game Description

Step into the dangerous world of organized crime in Mob Boss. In this thrilling idle clicker game, you'll start as a small-time boss with big dreams and your goal is to climb the criminal ladder by stealing, threatening, and expanding to become the most feared Mob Boss in the city!

## How do I play?

[Click here](https://zachplatypus.github.io/Mob-Boss/)

## Credits

Thank you to:

[AvKov](https://avkov.itch.io/city-tilemap-32x32) for the city assets that are used for this game.

[ankousse26](https://ankousse26.itch.io/pixel-art-boutons) for the pixel art buttons used for this game.

[Jestan](https://jestan.itch.io/weapons-pack) for the gun assets used for this game.

[Denzhn](https://denzhn.itch.io/pixelart-mafia-characters) for the mafia assets used for this game.

[Suit icons created by Nikita Golubev - Flaticon](https://www.flaticon.com/free-icons/suit)

& most of all, thank you to my teachers! Gerard, Cody, Caleb, and every one else has been a major major help through this project.

CHANGELOG.md
# Mob Boss Changelog

### v1.02
Hides all other elements upon victory

### v1.01
Bug fixes

### v1.00 (Release)
Added win conditions, just in time for presentation!

### v0.11 (Production Demo)
Bug fixes

### v0.10 (Presentation Demo)
Fully finished game, minus victory screen. Bodega & Factory functional all completed. Local Storage completed. Thanks a ton Gerard!

### v0.09
Minor bug fixes, finished functionality of SMG upgrade & robbing factory

### v0.08
Fixed Local Storage support

### v0.07
Added Local Storage support. Known Bug: bodegaManClicked & ownsSMG are not saving in Local Storage. Unsure as to why.

### v0.06
Fully constructed bodega gameplay loop. Thanks Gerard!

### v0.05
Majority of functionality for bodegas implemented. Several bugs, mostly relating to hiding bodegaMan

### v0.04
Major JS advancements, fully created initial functionality

### v0.03
Began JS development

### v0.02
Installed asset packs

### v0.01
Initialized Git, began development

assets/main.js
let cash = 0;
const cashDisplay = document.getElementById("cashDisplay");
const description = document.getElementById("description");
let bodegaCash = 15;
let factoryCash = 80;
let lastRobberyTime = 0;
cashDisplay.textContent = cash;
const buySMG = document.getElementById("gunSMG");
let ownsSMG = false;

// Local Storage set up

document.addEventListener('DOMContentLoaded', () => {
    const savedCash = localStorage.getItem('savedCash');
    if (savedCash !== null){
        console.log(savedCash)
        cash = parseInt(savedCash);
        cashDisplay.textContent = cash;
    };

    const savedBodegaMan = localStorage.getItem('savedBodegaMan');
    console.log(savedBodegaMan)
    if (savedBodegaMan === 'true'){
        bodegaManClicked = true;
        setInterval(autoBodega, 5000);
    } else {
        showBodegaMan();
    }; 

    const savedSMG = localStorage.getItem('savedSMG');
    console.log(savedSMG)
    if (savedSMG === 'true'){
        ownsSMG = true;
        gunFactory.style.display = 'inline-block';
        hideSMG();
    };

    const savedFactoryMan = localStorage.getItem('savedFactoryMan');
    console.log(savedFactoryMan)
    if (savedFactoryMan === 'true'){
        factoryManClicked = true;
        setInterval(autoFactory, 5000);
    } else {
        showFactoryMan();
    };
});

// Updating Cash & Victory Screen

const gameContainer = document.querySelector(".game-container"); // thank you Gerard!
function updateCash(){
    if (cash >= 5000){
        gameContainer.style.backgroundImage = "url(./assets/img/CityBGWin.png)"
        hideSMG();
        hideBodegaMan();
        hideFactoryMan();
        bodega.style.display = 'none';
        factory.style.display = 'none';
        updateCash();
    } else {
        cashDisplay.textContent = cash;
        localStorage.setItem('savedCash', cash);
        showBodegaMan();
        showFactoryMan();
    }
};

// Upgrade to an SMG

function hideSMG(){
    buySMG.style.display = 'none';
};

function upgradeSMG(){
    if (cash >= 500){
        if (ownsSMG === false){
        cash -= 500;
        updateCash();
        hideSMG();
        ownsSMG = true;
        gunFactory.style.display = 'inline-block';
        localStorage.setItem('savedSMG', ownsSMG);
        }
    }
};

buySMG.addEventListener("click", upgradeSMG);

buySMG.addEventListener("mouseover", () => {
    description.textContent = `Upgrade to an SMG, allowing you to perform higher level crime. Costs $500.`;
    buySMG.style.cursor = 'pointer';
});
buySMG.addEventListener("mouseout", () => {
    description.textContent = "";
});

// Reset Progress
const clearCashBtn = document.getElementById("clear");
clearCashBtn.addEventListener("click", clearCash);
function clearCash(){
    cash = 0;
    bodegaManClicked = false;
    localStorage.setItem('savedBodegaMan', bodegaManClicked);
    ownsSMG = false;
    localStorage.setItem('savedSMG', ownsSMG);
    factoryManClicked = false;
    localStorage.setItem('savedFactoryMan', factoryManClicked);
    updateCash();
};

// For demonstration purposes
const cashBtn = document.getElementById("cashBtn");
cashBtn.addEventListener("click", getCash);
function getCash(){
    cash += 500;
    updateCash();
}

assets/buttons.js
const description = document.getElementById("description")

// Step 1 of game, rob a bodega!
const bodega = document.querySelector(".bodega")
bodega.addEventListener("mouseover", () => {
    description.textContent = `Rob a Bodega, earning you $${bodegaCash}.`;
})
bodega.addEventListener("mouseout", () => {
    description.textContent = "";
})
function robBodega(){
    cash += bodegaCash;
    cashDisplay.textContent = cash;
    showBodegaMan();

    // HELP!!!!!!

    /*if (bodegaManClicked = true){
        cash += bodegaCash;
        cashDisplay.textContent = cash;
        showBodegaMan();
        console.log('test');
    } else {
        cash += bodegaCash;
        cashDisplay.textContent = cash;
        hideBodegaMan();
    }*/
}
bodega.addEventListener("click", robBodega);

// Hire a guy to hit up the bodega every now and then
const bodegaMan = document.getElementById("bodegaMan");
let bodegaManClicked = false;

// Only shows the man when you can afford the $100 for him.
function showBodegaMan(){
    if (cash >= 100){
        bodegaMan.style.display = 'inline-block';
    } else {
        bodegaMan.style.display = 'none';
    }
}

function hideBodegaMan(){
    bodegaMan.style.display = 'none';
}

// Bodega Man does stuff!
bodegaMan.addEventListener("mouseover", function(){
    description.textContent = `Hire an associate to handle the bodega for you, earning you $${bodegaCash} every ${robberyTime} seconds.`;
})
bodegaMan.addEventListener("mouseout", function(){
    description.textContent = "";
})
function autoBodega(){
    setInterval(autoBodega, 5000)
    cash += bodegaCash;
    cashDisplay.textContent = cash;
}
bodegaMan.addEventListener("click", () => {
    autoBodega();
    bodegaManClicked = true;
    hideBodegaMan();
    cash -= 100;
    cashDisplay.textContent = cash;
})

// bodegaCash is doubling each time. don't know why??

assets/factory.js
// Step 1 of game, rob a factory!
const factory = document.querySelector(".factory")
let factoryInterval = 3000;

factory.addEventListener("mouseover", () => {
    description.textContent = `Rob a Factory, earning you $${factoryCash}.`;
    factory.style.cursor = 'pointer';
});
factory.addEventListener("mouseout", () => {
    description.textContent = "";
});

function robFactory(){
    const currentTime = Date.now();
    const timeSinceLastRobbery = currentTime - lastRobberyTime;
    if (ownsSMG === true){
        if (timeSinceLastRobbery >= factoryInterval){
            cash += factoryCash;
            updateCash();
            lastRobberyTime = currentTime;
            console.log("+$80");
        }
    }
};

factory.addEventListener("click", robFactory);

// Hire a guy to hit up the factory every now and then
const factoryMan = document.getElementById("factoryMan");

// Only shows the man when you can afford the $100 for him.
let factoryManClicked = false;

function showFactoryMan(){
    if (cash >= 1000){
        if (factoryManClicked === false){
            factoryMan.style.display = 'inline-block';
        }
    } else {
        factoryMan.style.display = 'none';
    }
};

function hideFactoryMan(){
    factoryMan.style.display = 'none';
};

// Factory Man does stuff!
factoryMan.addEventListener("mouseover", function(){
    description.textContent = `Hire an associate to handle the factory for you, earning you $${factoryCash} every 5 seconds. Costs $1000.`;
    factoryMan.style.cursor = 'pointer';
});

factoryMan.addEventListener("mouseout", function(){
    description.textContent = "";
});

function autoFactory(){
    cash += factoryCash;
    updateCash();
    showFactoryMan();
    console.log("+$80");
};

factoryMan.addEventListener("click", () => {
    setInterval(autoFactory, 5000);
    factoryManClicked = true;
    hideFactoryMan();
    cash -= 1000;
    updateCash();
    localStorage.setItem('savedFactoryMan', factoryManClicked);
});



assets/bodega.js
// Step 1 of game, rob a bodega!
const bodega = document.querySelector(".bodega")
let bodegaInterval = 1500;

bodega.addEventListener("mouseover", () => {
    description.textContent = `Rob a Bodega, earning you $${bodegaCash}.`;
    bodega.style.cursor = 'pointer';
});
bodega.addEventListener("mouseout", () => {
    description.textContent = "";
});

function robBodega(){
    const currentTime = Date.now();
    const timeSinceLastRobbery = currentTime - lastRobberyTime;
    if (timeSinceLastRobbery >= bodegaInterval){
        cash += bodegaCash;
        updateCash();
        lastRobberyTime = currentTime;
        console.log("+$15");
        }
};

bodega.addEventListener("click", robBodega);

// Hire a guy to hit up the bodega every now and then
const bodegaMan = document.getElementById("bodegaMan");

// Only shows the man when you can afford the $100 for him.
let bodegaManClicked = false;

function showBodegaMan(){
    if (cash >= 100){
        if (bodegaManClicked === false){
            bodegaMan.style.display = 'inline-block';
        }
    } else {
        bodegaMan.style.display = 'none';
    }
};

function hideBodegaMan(){
    bodegaMan.style.display = 'none';
};

// Bodega Man does stuff!
bodegaMan.addEventListener("mouseover", () => {
    description.textContent = `Hire an associate to handle the bodega for you, earning you $${bodegaCash} every 3 seconds. Costs $100.`;
    bodegaMan.style.cursor = 'pointer';
});

bodegaMan.addEventListener("mouseout", () => {
    description.textContent = "";
});

function autoBodega(){
    cash += bodegaCash;
    updateCash();
    showFactoryMan();
    console.log("+$15");
};

bodegaMan.addEventListener("click", () => {
    setInterval(autoBodega, 3000);
    bodegaManClicked = true;
    hideBodegaMan();
    cash -= 100;
    updateCash();
    localStorage.setItem('savedBodegaMan', bodegaManClicked);
});